src/combat.ts

```diff
@@ -170,51 +176,47 @@ export class Macro extends StrictMacro {
     switch (itemOrSkill) {
       case $item`Spooky Putty sheet`:
         return this.externalIf(
-          get("spookyPuttyCopiesMade") + Math.max(1, get("_raindohCopiesMade")) < 6,
+          get("spookyPuttyCopiesMade") + Math.max(1, get("_raindohCopiesMade")) < 6 &&
+            $items`Spooky Putty sheet, Spooky Putty monster`.some((item) => have(item)),
           Macro.tryItem(itemOrSkill)
         );
       case $item`Rain-Doh black box`:
         return this.externalIf(
-          get("_raindohCopiesMade") + Math.max(1, get("spookyPuttyCopiesMade")) < 6,
-          Macro.tryItem(itemOrSkill)
+          get("_raindohCopiesMade") + Math.max(1, get("spookyPuttyCopiesMade")) < 6 &&
+            $items`Rain-Doh black box, Rain-Doh box full of monster`.some((item) => have(item)),
+          Macro.tryHaveItem(itemOrSkill)
```

Comment:
this one needs to be tryItem, because the macro is generated before the combat starts and you may be just about to use a box full of monster and won't have the black box in your inventory yet

Reply:
good catch; i had reverted putty but not doh